javascript/javascript2/week2/movies_task.js

```diff
@@ -0,0 +1,138 @@
+'use strict'
+console.log(movies.length);
+
+// 1.Create an array of movies containing the movies with a short title (you define what short means)
+const shortTitleLength = 5;
+const shortMovieTitles = movies.filter(movie => movie.title.length <= shortTitleLength);
+console.log(shortMovieTitles);
+
+// 2.Create an array of movie titles with long movie titles
+const longTitleLength = 30;
+const longMovieTitles = movies.filter(movie => movie.title.length >= longTitleLength)
+console.log(longMovieTitles);
+
+// 3.Count the number of movies made between 1980-1989 (including both the years)
+const eightiesMovies = movies.filter(movie => movie.year >= 1980 && movie.year <= 1989);
+console.log(eightiesMovies);
+
+// 4.Create a new array that has an extra key called tag.
+// The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const movieRatings = movies.map(movie => {
+    if (movie.rating >= 7) {
+        movie.tag = 'Good';
+    } else if (movie.rating >= 4 && movie.rating < 7) {
+        movie.tag = 'Average';
+    } else {
+        movie.tag = 'Bad';
+    }
+    return movie;
+})
+console.log(movieRatings);
+
+// 5. Using chaining, first filter the movies array to only contain the movies rated higher than 6.
+// Now map the movies array to only the rating of the movies.
+
+const highRatedMovies = movies
+    .filter(movie => movie.rating > 6)
+    .map(movie => movie.rating);
+console.log(highRatedMovies);
+
+// 6. Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin.
+// So if there were 3 movies that contained Surfer, 1 with Alien and 2 with Benjamin, you would return 6.
+// Can you make sure the search is case insensitive?
+
+const keyWord1 = 'Surfer'.toLowerCase();
+const keyWord2 = 'Alien'.toLowerCase();
+const keyWord3 = 'Benjamin'.toLowerCase();
+
+const sumMoviesWithKeyWords = movies.filter(movie => {
+    const titleWords = movie.title.toLowerCase().split(' ');
+    return (titleWords.includes(keyWord1) || titleWords.includes(keyWord2) || titleWords.includes(keyWord3))
+}) ;
+console.log(sumMoviesWithKeyWords);
+
+// 7.Create an array of movies where a word in the title is duplicated. Fx "Star Wars: The Clone Wars" the word Wars is duplicated.
+// Here are some madeup examples of movies with duplicated words in the title: "The three men and the pistol", "Chase three - The final chase"
+
+const duplicatedWordMovies = movies.filter(movie => {
+    const titleWords = movie.title.toLowerCase().split(' ');
+    return hasDuplicateWords(titleWords);
+})
+
+function hasDuplicateWords(words) {
+    for (let i = 0; i < words.length - 1; i++) {
+        for (let j = i + 1; j < words.length; j++) {
+            if (words[i] === words[j]) {
```

Comment:
try to avoid loop with in the loop
const duplicatedWordMovies = movies.filter(movie =>{
    const titleWords = movie .title.toLowerCase().split(" ");
    const repeatingWords = titleWords .filter((word,wordIndex)=>{
    return titleWords .indexOf(word) !== wordIndex
    });
    return repeatingWords .length >0;
})